packages-exp/auth-exp/src/core/user/user_impl.ts

```diff
@@ -102,6 +106,28 @@ export class UserImpl implements User {
     return reload(this);
   }
 
+  private reloadUserInfo: APIUserInfo | null = null;
+  private reloadListener: NextFn<APIUserInfo> | null = null;
+
+  _onReload(callback: NextFn<APIUserInfo>): void {
+    // There should only ever be one listener, and that is a single instance of MultiFactorUser
+    assert(!this.reloadListener, this.auth.name);
+    this.reloadListener = callback;
+    if (this.reloadUserInfo) {
+      this._notifyReloadListener(this.reloadUserInfo);
+      this.reloadUserInfo = null;
+    }
+  }
+
+  _notifyReloadListener(userInfo: APIUserInfo): void {
+    if (this.reloadListener) {
+      this.reloadListener(userInfo);
```

Comment:
why not set `this.reloadUserInfo = null` here instead of on line 118?

Reply:
I prefer the check next to the use, that way the logic is check, use, null-out, we don't actually use reloadUserInfo in _notifyReloadListener so it seems a bit off to null it out without using it